pkg/proxy/iptables/proxier.go

```diff
@@ -975,6 +975,16 @@ func (proxier *Proxier) syncProxyRules() {
 	if err != nil {
 		klog.ErrorS(err, "Failed to get node ip address matching nodeport cidrs, services with nodeport may not work as intended", "CIDRs", proxier.nodePortAddresses)
 	}
+	// nodeAddresses may contain dual-stack zero-CIDRs if proxier.nodePortAddresses is empty.
+	// Ensure nodeAddresses only contains the addresses for this proxier's IP family.
```

Comment:
yeah, this is the problem, we are not filtering by the ip family, but I think that this should be extended for all the IPs, not only the zero ones ...

Reply:
https://github.com/kubernetes/kubernetes/blob/7846bbbda0ab921de47292b79722dc7a5420e7c8/pkg/proxy/iptables/proxier.go#L1292 
As the above code declares, `nodeAddresses` is supposed to contain single IP family only, which is true for most cases as `proxier.nodePortAddresses` has been filtered by IP family, so `GetNodeAddresses` returns single IP family only in that case:
https://github.com/kubernetes/kubernetes/blob/7846bbbda0ab921de47292b79722dc7a5420e7c8/pkg/proxy/iptables/proxier.go#L291-L292

The exception is when `proxier.nodePortAddresses` is empty, it returns dual-stack zero-CIDRs, as the new comment explains.

Reply:
ah, great, I saw the comment explaining that but didn't find where those addresses were actually filtered
Thanks